File: cpp/minmax.cpp
/*
 * minmax.cpp
 */

#include <iostream>
//#include <cstdlib>
//#include <ctime>
#include <windows.h>
#include <time.h>

using namespace std;

//int losuj(int ile, int zakres);

int main(int argc, char **argv)
{
    int ile = 0;
    int zakres = 0;
    int lista[zakres];
    int max = lista[0];
    int min = lista[0];
    
    
    cout<<"Ile liczb: ";
    cin>>ile;
    cout<<"Zakres: ";
    cin>>zakres;
    
     srand(time(NULL));
     
    for(int i=0; i<=ile;i++)
    {
        lista[i] = rand()%zakres+1;
        
        cout<<lista[i]<<endl;
    }
    for(int i=0; i<ile;i++)
    {
        if(lista[i] > max)
        {
            max = lista[i];
        }
        if(lista[i] < min)
        {
            min = lista[i];
        }
    }
    
    cout<<"Maksymalna= "<<max<<endl;
    cout<<"Minimalna= "<<min<<endl;
    
    return 0;
}

/*
int losuj(int ile, int zakres)
{
    int lista[zakres];
    
    for(int i=0; i<ile;i++)
    {
        lista[i] = rand()%zakres+1;
        
        cout<<lista[i]<<endl;
    }
    
    return lista[];
}
*/

File: python/minmax.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
from random import randint


def minimum(lista):
    min = lista[0]
    for i in lista:
        if i < min:
            min = i
    return min


def maksimum(lista):
    max = lista[0]
    for i in lista:
        if i > max:
            max = i
    return max


def minmax(lista):
    lmax = []
    lmin = []
    index = 0

    for i in range(int(len(lista) / 2)):
        if lista[index] > lista[index + 1]:
            lmax.append(lista[index])
            lmin.append(lista[index + 1])
        else:
            lmin.append(lista[index])
            lmax.append(lista[index + 1])
        index += 2

    print("Maksymalna: ", maksimum(lmax))
    print("Minimalna: ", minimum(lmin))

    return 0


def losuj(ile, zakres):
    lista = []
    for i in range(ile):
        lista.append(randint(0, zakres))
    return lista


def main(args):
    ile = int(input("Ile liczb: "))
    zakres = int(input("Podaj zkres: "))

    assert minimum([6, 8, 0, 9, 1, ]) == 0
    assert maksimum([6, 8, 0, 9, 1, ]) == 9

    lista = losuj(ile, zakres)
    print(lista)

    minmax(lista)

    # print("Najmniejsza: ", minimum(lista))
    # print("Najwieksza: ", maksimum(lista))

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
